Database/CleanArchive.py

```diff
@@ -88,12 +96,69 @@ def clean(all=True):
         transaction = {
             '$pull': {'snippets': {'transcript': ''}}
         }
+        if duplicate < 1.0:
+            print('Duplicate found')
+            set_fields['duplicate'] = True
+            duplicates['metadata.Subtitle'] += 1
+        else:
+            set_fields['duplicate'] = False
         if errors:
             set_fields['errors'] = errors
         else:
             transaction['$unset'] = {'errors': 1}
         if set_fields:  # $set cannot be empty.
             transaction['$set'] = set_fields
+
+
+
         db.ArchiveIndex.update_one({'_id': episode['_id']}, transaction)
     print('Updates:', updates)
     print('Failures:', failures)
+    print('Duplicates', duplicates)
+
+
+def checkDuplicate(epi):
+    upperBound = db.ArchiveIndex.find({'identifier': epi['identifier']})[0]['date']
+    dateTimeObj = datetime.datetime.strptime(upperBound, '%Y-%m-%dT%H:%M:%SZ')
+    lowerBound = str(dateTimeObj - dateTimeObj - datetime.timedelta(days=1))
+    fields = db.ArchiveIndex.find({'date': {'$lte': upperBound, '$gte': lowerBound}}, {'snippets': 1})
+    currentEpisode = ''
+    for snip in epi['snippets']:
+        currentEpisode += str(snip['transcript'])
+    for field in fields:
+        compareEpisode = ''
+        cosine = 0
+        try:
```

Comment:
Avoid try/except blocks where possible. I think it should be in the main cleaning function like the other try/except block so the error can be logged for the user.

Reply:
It is nice to have down there because if one episode can't be compared it will still compare currentEpisodes to all the others whereas a try catch in the main function will catch and quit after one error right?

Reply:
Hmm true. If you add to your query $exists: Datetime_UTC, then I'd expect there to be no errors. At least no errors caused by missing fields.